### PR TITLE
Add new source: rawlh

### PR DESCRIPTION
**Pull request by Pavkazzz**
http://rawlh.com/

Basic functionality

### CODE REVIEW

**Comment by CarlosEsco** [comment]
Please add icons

File: src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+    override fun chapterListSelector() = " table.table.table-hover tbody tr"
+
+    override fun chapterFromElement(element: Element): SChapter {
+        val urlElement = element.select("td a").first()
+
+        val chapter = SChapter.create()
+        chapter.setUrlWithoutDomain("/" + urlElement.attr("href"))
+        chapter.name = urlElement.text()
+//        TODO:
```
Comment (CarlosEsco) [inline src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt]: remove commented out code

File: src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            }
+        }
+        return pages
+    }
+
+    override fun imageUrlParse(document: Document) = ""
+
+
+
+//    private class OrderBy : Filter.Sort("Сортировка",
```
Comment (CarlosEsco) [inline src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt]: same here

File: src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+
+    // TODO: Country
+    override fun getFilterList() = FilterList(
+            Status(),
+            GenreList(getGenreList())
+    )
+
+
+    /* [...document.querySelectorAll("div.panel-body a")].map((el,i) =>
```
Comment (CarlosEsco) [inline src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt]: same

File: src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+
+class Rawlh : ParsedHttpSource() {
+
+    override val name = "RawLH"
+
+    override val baseUrl = "http://rawlh.com"
+
+    override val lang = "en"
```
Comment (paronos) [inline src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt]: `lang = "ja"`

File: src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt [hunk 5]
```diff
@@ -0,0 +1,182 @@
+package eu.kanade.tachiyomi.extension.en.rawlh
```
Comment (paronos) [inline src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt]: `package eu.kanade.tachiyomi.extension.ja.rawlh`

File: src/en/rawlh/build.gradle [hunk 6]
```diff
@@ -0,0 +1,13 @@
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+
+ext {
+    appName = 'Tachiyomi: Rawlh'
+    pkgNameSuffix = "en.rawlh"
```
Comment (paronos) [inline src/en/rawlh/build.gradle]: `pkgNameSuffix = "ja.rawlh"`

File: src/en/rawlh/build.gradle [hunk 7]
```diff
@@ -0,0 +1,13 @@
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+
+ext {
+    appName = 'Tachiyomi: Rawlh'
```
Comment (paronos) [inline src/en/rawlh/build.gradle]: `appName = 'Tachiyomi: RawLH'`

File: src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+//            arrayOf("Дата", "Популярность", "Имя", "Главы"),
+//            Filter.Sort.Selection(1, false))
+
+
+    private class Status : Filter.Select<String>("Status", arrayOf("Any", "Ongoing", "Completed"))
+    private class GenreList(genres: List<Genre>) : Filter.Group<Genre>("Genre", genres)
+    private class Genre(name: String, val id: String = name.replace(' ', '_')) : Filter.TriState(name)
+
+
+    // TODO: Country
```
Comment (paronos) [inline src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt]: `// TODO: Country, Author, Group`
(better to add them actually)

File: src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+
+
+//    private class OrderBy : Filter.Sort("Сортировка",
+//            arrayOf("Дата", "Популярность", "Имя", "Главы"),
+//            Filter.Sort.Selection(1, false))
+
+
+    private class Status : Filter.Select<String>("Status", arrayOf("Any", "Ongoing", "Completed"))
+    private class GenreList(genres: List<Genre>) : Filter.Group<Genre>("Genre", genres)
+    private class Genre(name: String, val id: String = name.replace(' ', '_')) : Filter.TriState(name)
```
Comment (paronos) [inline src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt]: it seems to me that the site uses `'+'` instead of `'_'` in the search url.

Comment (Pavkazzz) [reply src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt]: Thx, but this functionality seems to be broken

File: src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt (hunk 4 above)
Comment (Pavkazzz) [reply src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt]: Why?
What does it means?

Comment (paronos) [reply src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt]: This specifies that the manga are in Japanese (instead of English).

**Comment by CarlosEsco** [comment]
Since they don't have an icon use the orange with white text and generate with this https://romannurik.github.io/AndroidAssetStudio/icons-launcher.html#foreground.type=clipart&foreground.clipart=android&foreground.space.trim=1&foreground.space.pad=0.25&foreColor=rgba(96%2C%20125%2C%20139%2C%200)&backColor=rgb(68%2C%20138%2C%20255)&crop=0&backgroundShape=square&effects=none&name=ic_launcher

File: src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt (hunk 9 above)
Comment (CarlosEsco) [reply src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt]: So either fix it or remove it

File: src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt (hunk 3 above)
Comment (CarlosEsco) [reply src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt]: Sorry I meant to add this for all the commented out code.

Comment (CarlosEsco) [reply src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt]: Sorry I meant to add this for all the commented out code.

**Comment by Pavkazzz** [comment]
Wow Carlos, u are amazing! Thx!!!

File: src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt (hunk 4 above)
Comment (Pavkazzz) [reply src/en/rawlh/src/eu/kanade/tachiyomi/extension/en/rawlh/Rawlh.kt]: Oh my god, it's japanese source 😄 
My bad! Thx

**Comment by Pavkazzz** [comment]
Thx for review
Ready to merge!

**Pull request merged**